trin-core/src/portalnet/types/messages.rs

```diff
@@ -19,6 +19,66 @@ use crate::portalnet::{types::uint::U256, Enr};
 
 pub type ByteList = VariableList<u8, typenum::U2048>;
 
+/// Custom payload element of Ping and Pong overlay messages
+#[derive(Debug, PartialEq, Clone)]
+pub struct CustomPayload {
+    payload: ByteList,
+}
+
+impl TryFrom<&Value> for CustomPayload {
+    type Error = ValidationError;
+
+    fn try_from(value: &Value) -> Result<Self, Self::Error> {
+        let value = value
+            .as_str()
+            .ok_or_else(|| ValidationError::new("Empty custom payload value"))?;
```

Comment:
I might be confused here, but shouldn't this error message be more along the lines of "this value is not a string"? `Value` is an enum, and I guess we'd want to display if deserialization returned variants other than `String` (which would also catch any `Null` cases)

Reply:
You are right, we can use a better message. This was copy-paste from https://github.com/ethereum/trin/blob/73f38b26c1177aa7326ebc91b6503a2257e64e3d/trin-core/src/portalnet/types/messages.rs#L464 :)

Reply:
Hahah, yeah looks like it was my crappy error message - mind updating that one as well here?